internal/credentials/xds/handshake_info.go

```diff
@@ -186,7 +242,18 @@ func (hi *HandshakeInfo) ServerSideTLSConfig(ctx context.Context) (*tls.Config,
 		if err != nil {
 			return nil, fmt.Errorf("xds: fetching trusted roots from CertificateProvider failed: %v", err)
 		}
-		cfg.ClientCAs = km.Roots
+		if km.SPIFFEBundleMap != nil {
+			// ClientAuth, if set above tls.RequireAnyClientCert, must be
```

Comment:
nit: Should we say "strictly above tls.RequireAnyClientCert"? Is this correct?

Reply:
I'll change to `greater than`